src/components/IdleModal.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import {
+  ModalContainer,
+  ModalBody,
+  Text,
+  Button,
+  InjectedModalProps,
+  ModalTitle,
+  Heading,
+  ModalHeader,
+} from '@pancakeswap/uikit'
+import { useTranslation } from 'contexts/Localization'
+import styled from 'styled-components'
+
+const GradientModalHeader = styled(ModalHeader)`
+  background: ${({ theme }) => theme.colors.gradients.bubblegum};
+  padding-bottom: 24px;
+  padding-top: 24px;
+`
+
+interface IdleModalProps extends InjectedModalProps {
+  onSuccess: () => void
+}
+
+const IdleModal: React.FC<IdleModalProps> = ({ onSuccess, onDismiss }) => {
+  const { t } = useTranslation()
+
+  const handleConfirm = () => {
+    onSuccess()
+    onDismiss()
+  }
+
+  return (
+    <ModalContainer title={t('Oops!')} minWidth="320px">
+      <GradientModalHeader>
+        <ModalTitle>
+          <Heading scale="lg">{t('Oops!')}</Heading>
+        </ModalTitle>
+      </GradientModalHeader>
+      <ModalBody p="24px" maxWidth="400px">
+        <Text as="p" mb="16px">
+          {t('You have been idle for a while')}
+        </Text>
+        <Text as="p" mb="16px">
+          {t('You will get timed out. You want to stay?')}
+        </Text>
+        <Button width="100%" onClick={handleConfirm} mb="16px">
+          {t('Continue')}
+        </Button>
+      </ModalBody>
+    </ModalContainer>
+  )
+}
```

Comment:
I don't think this modal is necessary. We can pause the fetching when the user is idle or tab inactive. As I mentioned in my other comment let's start with just pausing `useRefresh` when the tab is inactive.

Reply:
I removed idle modal and make tab active set to to true in tabActivity hook.

Reply:
> I don't think this modal is necessary. We can pause the fetching when the user is idle or tab inactive. As I mentioned in my other comment let's start with just pausing `useRefresh` when the tab is inactive.

If we put that check in useRefresh, we might have consequences for example cake price in page title showing an old value when tab is hidden or waiting a bit to get updated state in prediction (which might be frustrating for the user).